ansible/roles/prosody/files/mod_muc_get_media_type.lua

```diff
@@ -0,0 +1,68 @@
+---
+--- Generated by Luanalysis
+--- Created by ltorje.
+--- DateTime: 06.02.2024 16:43
+---
+--- no regex, less logging, check room first and exist if set already
+--- getting the room --> e.g. <iq xmlns="jabber:client" id="26bbe412-6dde-4225-8ca8-0690a462b89d:sendIQ" to="borisyana1to1@conference.meet.jit.si/focus"
+--- room = borisyana1to1@conference.meet.jit.si
+
+local json = require "cjson";
+local jid = require "util.jid";
+local get_room_from_jid = module:require "util".get_room_from_jid;
+
+function handle_media_event(event)
+    local stanza = event.stanza;
+    if stanza.name == "iq" then
+        local room_jid = jid.bare(stanza.attr.to);
+        local room = get_room_from_jid(room_jid);
+
+        if room == nil then
+            return
+        end
+
+        if room.had_video then
+            return
+        end
+
+        local mediaType = "audio";
+
+        -- https://xmpp.org/extensions/xep-0339.html
+        local jingle_source_add = stanza:get_child_with_attr("jingle", "urn:xmpp:jingle:1", "action", "source-add");
+        if jingle_source_add then
+            mediaType = jingle_source_add:find("content/{urn:xmpp:jingle:apps:rtp:1}description/{urn:xmpp:jingle:apps:rtp:ssma:0}source@videoType");
+        else
+            -- https://xmpp.org/extensions/xep-0166.html#def-action-session-accept
+            local jingle_session_accept = stanza:get_child_with_attr("jingle", "urn:xmpp:jingle:1", "action", "session-accept");
```

Comment:
I think this is sufficient for the jicofo case, but for p2p you need to handle:

* session-initiate to catch the case when the p2p initiator has video (note this should NOT be handled for the jicofo case)
* the case where the content 'name' and description 'media' are arbitrary (e.g. "0" and "1") instead of "audio" and "video".

There's a detailed description and example messages in the document I shared

Reply:
It turns out the conference always starts in jvb mode, not p2p, so it is sufficient to monitor session-accept to jicofo plus source-add to anyone